pkg/controller/syncset/syncset_controller_test.go

```diff
@@ -0,0 +1,313 @@
+package syncset
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	constraintclient "github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/drivers/rego"
+	syncsetv1alpha1 "github.com/open-policy-agent/gatekeeper/v3/apis/syncset/v1alpha1"
+	cm "github.com/open-policy-agent/gatekeeper/v3/pkg/cachemanager"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/controller/config/process"
+	syncc "github.com/open-policy-agent/gatekeeper/v3/pkg/controller/sync"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/fakes"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/readiness"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/syncutil"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/target"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/watch"
+	testclient "github.com/open-policy-agent/gatekeeper/v3/test/clients"
+	"github.com/open-policy-agent/gatekeeper/v3/test/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	configMapGVK = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "ConfigMap"}
+	nsGVK        = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Namespace"}
+	podGVK       = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Pod"}
+)
+
+const (
+	timeout = time.Second * 10
+	tick    = time.Second * 2
+)
+
+// Test_ReconcileSyncSet_wConfigController verifies that SyncSet and Config resources
+// can get reconciled and their respective specs are added to the data client.
+func Test_ReconcileSyncSet_wConfigController(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	require.NoError(t, testutils.CreateGatekeeperNamespace(cfg))
+
+	tr := setupTest(ctx, t, false, true)
+	mgr := *tr.mgr
+	c := tr.c
+	cfClient := tr.cfClient
+
+	instanceConfig := testutils.ConfigFor([]schema.GroupVersionKind{})
+	instanceSyncSet1 := &syncsetv1alpha1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "syncset1",
+		},
+	}
+	instanceSyncSet2 := &syncsetv1alpha1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "syncset2",
+		},
+	}
+	configMap := testutils.UnstructuredFor(configMapGVK, "", "cm1-name")
+	pod := testutils.UnstructuredFor(podGVK, "", "pod1-name")
+
+	// for sync controller
```

Comment:
"for sync controller" doesn't mean much.  What are we doing here?  Seems more like:

```
// Install sync controller
```